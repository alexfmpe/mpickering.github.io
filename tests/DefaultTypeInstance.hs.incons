[(tests/examples/DefaultTypeInstance.hs:6:8-34,
  (AnnEqual, [tests/examples/DefaultTypeInstance.hs:6:31]))]