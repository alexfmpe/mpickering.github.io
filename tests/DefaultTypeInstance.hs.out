{-# LANGUAGE TypeFamilies #-}


class Foldable t where
  type FoldableConstraint t x :: Constraint
  type FoldableConstraint t x   ()

==============
tests/examples/DefaultTypeInstance.hs
==============
lengths:(134,134)

==============

({ tests/examples/DefaultTypeInstance.hs:1:1 }
 Just (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnEofPos),DP (1,0))])
 (HsModule 
  (Nothing) 
  (Nothing) 
  [] 
  [
   ({ tests/examples/DefaultTypeInstance.hs:(4,1)-(6,34) }
    Just (Ann (DP (3,1)) (ColDelta 1) DP (3,1) [DComment (DP (0,0),DP (0,29)) "{-# LANGUAGE TypeFamilies #-}"] [((G AnnClass),DP (0,0)),((G AnnWhere),DP (0,1))])
    (TyClD 
     (ClassDecl 
      ({ <no location info> }
       Just (Ann (DP (-4,0)) (ColDelta 0) DP (-4,0) [] [])
       []) 
      ({ tests/examples/DefaultTypeInstance.hs:4:7-14 }
       Just (Ann (DP (0,1)) (ColDelta 7) DP (0,1) [] [((G AnnVal),DP (0,0))])
       (Unqual {OccName: Foldable})) 
      (HsQTvs 
       [] 
       [
        ({ tests/examples/DefaultTypeInstance.hs:4:16 }
         Just (Ann (DP (0,1)) (ColDelta 16) DP (0,1) [] [((G AnnVal),DP (0,0))])
         (UserTyVar 
          (Unqual {OccName: t})))]) 
      [] 
      [] {Bag(Located (HsBind RdrName)): 
      []} 
      [
       ({ tests/examples/DefaultTypeInstance.hs:5:3-43 }
        Just (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnType),DP (0,0)),((G AnnDcolon),DP (0,1))])
        (FamilyDecl 
         (OpenTypeFamily) 
         ({ tests/examples/DefaultTypeInstance.hs:5:8-25 }
          Just (Ann (DP (0,1)) (ColDelta 8) DP (0,1) [] [((G AnnVal),DP (0,0))])
          (Unqual {OccName: FoldableConstraint})) 
         (HsQTvs 
          [] 
          [
           ({ tests/examples/DefaultTypeInstance.hs:5:27 }
            Just (Ann (DP (0,1)) (ColDelta 27) DP (0,1) [] [((G AnnVal),DP (0,0))])
            (UserTyVar 
             (Unqual {OccName: t}))),
           ({ tests/examples/DefaultTypeInstance.hs:5:29 }
            Just (Ann (DP (0,1)) (ColDelta 29) DP (0,1) [] [((G AnnVal),DP (0,0))])
            (UserTyVar 
             (Unqual {OccName: x})))]) 
         (Just 
          ({ tests/examples/DefaultTypeInstance.hs:5:34-43 }
           Just (Ann (DP (0,1)) (ColDelta 34) DP (0,1) [] [((G AnnVal),DP (0,0))])
           (HsTyVar 
            (Unqual {OccName: Constraint}))))))] 
      [
       ({ tests/examples/DefaultTypeInstance.hs:6:3-34 }
        Just (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnType),DP (0,0))])
        (TyFamEqn 
         ({ tests/examples/DefaultTypeInstance.hs:6:8-25 }
          Just (Ann (DP (0,1)) (ColDelta 8) DP (0,1) [] [((G AnnVal),DP (0,0))])
          (Unqual {OccName: FoldableConstraint})) 
         (HsQTvs 
          [] 
          [
           ({ tests/examples/DefaultTypeInstance.hs:6:27 }
            Just (Ann (DP (0,1)) (ColDelta 27) DP (0,1) [] [((G AnnVal),DP (0,0))])
            (UserTyVar 
             (Unqual {OccName: t}))),
           ({ tests/examples/DefaultTypeInstance.hs:6:29 }
            Just (Ann (DP (0,1)) (ColDelta 29) DP (0,1) [] [((G AnnVal),DP (0,0))])
            (UserTyVar 
             (Unqual {OccName: x})))]) 
         ({ tests/examples/DefaultTypeInstance.hs:6:33-34 }
          Just (Ann (DP (0,3)) (ColDelta 33) DP (0,3) [] [((G AnnOpenP),DP (0,0)),((G AnnCloseP),DP (0,0))])
          (HsTupleTy 
           (HsBoxedOrConstraintTuple) 
           []))))] 
      [] 
      (PlaceHolder))))] 
  (Nothing) 
  (Nothing)))
==============
[(AnnKey tests/examples/DefaultTypeInstance.hs:1:1 CN "HsModule",
  (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnEofPos),DP (1,0))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:(4,1)-(6,34) CN "TyClD",
  (Ann (DP (3,1)) (ColDelta 1) DP (3,1) [DComment (DP (0,0),DP (0,29)) "{-# LANGUAGE TypeFamilies #-}"] [((G AnnClass),DP (0,0)),((G AnnWhere),DP (0,1))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:4:7-14 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 7) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:4:16 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 16) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:5:3-43 CN "FamilyDecl",
  (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnType),DP (0,0)),((G AnnDcolon),DP (0,1))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:5:8-25 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 8) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:5:27 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 27) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:5:29 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 29) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:5:34-43 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 34) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:6:3-34 CN "TyFamEqn",
  (Ann (DP (1,3)) (ColDelta 3) DP (1,3) [] [((G AnnType),DP (0,0))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:6:8-25 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 8) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:6:27 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 27) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:6:29 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 29) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/DefaultTypeInstance.hs:6:33-34 CN "HsTupleTy",
  (Ann (DP (0,3)) (ColDelta 33) DP (0,3) [] [((G AnnOpenP),DP (0,0)),((G AnnCloseP),DP (0,0))])),
 (AnnKey <no location info> CN "[]",
  (Ann (DP (-4,0)) (ColDelta 0) DP (-4,0) [] []))]
==============
([((tests/examples/DefaultTypeInstance.hs:(4,1)-(6,34), AnnClass),
   [tests/examples/DefaultTypeInstance.hs:4:1-5]),
  ((tests/examples/DefaultTypeInstance.hs:(4,1)-(6,34), AnnSemi),
   [tests/examples/DefaultTypeInstance.hs:7:1]),
  ((tests/examples/DefaultTypeInstance.hs:(4,1)-(6,34), AnnWhere),
   [tests/examples/DefaultTypeInstance.hs:4:18-22]),
  ((tests/examples/DefaultTypeInstance.hs:5:3-43, AnnDcolon),
   [tests/examples/DefaultTypeInstance.hs:5:31-32]),
  ((tests/examples/DefaultTypeInstance.hs:5:3-43, AnnSemi),
   [tests/examples/DefaultTypeInstance.hs:6:3]),
  ((tests/examples/DefaultTypeInstance.hs:5:3-43, AnnType),
   [tests/examples/DefaultTypeInstance.hs:5:3-6]),
  ((tests/examples/DefaultTypeInstance.hs:6:3-34, AnnType),
   [tests/examples/DefaultTypeInstance.hs:6:3-6]),
  ((tests/examples/DefaultTypeInstance.hs:6:8-34, AnnEqual),
   [tests/examples/DefaultTypeInstance.hs:6:31]),
  ((tests/examples/DefaultTypeInstance.hs:6:33-34, AnnCloseP),
   [tests/examples/DefaultTypeInstance.hs:6:34]),
  ((tests/examples/DefaultTypeInstance.hs:6:33-34, AnnOpenP),
   [tests/examples/DefaultTypeInstance.hs:6:33]),
  ((<no location info>, AnnEofPos),
   [tests/examples/DefaultTypeInstance.hs:7:1])],
 [(<no location info>,
   [AnnBlockComment "{-# LANGUAGE TypeFamilies #-}"])])